### PR TITLE
Shop 7695 files documents cannot be downloaded from notes if they contain a space in the file name

### PR DESCRIPTION
**Pull request by ghost**
Add a clean_file function to replace spaces with underscores.

### CODE REVIEW

**Comment by colinross** [comment]
@artempopkov0 Can you please include a spec for this change?

**Comment by ghost** [comment]
I add spec for ruby code changes, but not for js, because it can't be done, but we don't need it, because i got js code changes  from source gem code, so i think it already was tested.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -2,7 +2,6 @@
 require 's3_direct_upload'
 
 RSpec.configure do |config|
-  config.treat_symbols_as_metadata_keys_with_true_values = true
```
Comment (colinross) [inline spec/spec_helper.rb:5]: Why this change?

**Review by colinross — COMMENTED** [review]
was the spec helper change intended?

File: spec/spec_helper.rb (hunk 1 above)
Comment (ghost) [reply spec/spec_helper.rb:5]: Because right now this is true for default

**Comment by ghost** [comment]
> was the spec helper change intended?

Could you please clarify what do you mean ?

**Comment by colinross** [comment]
> > was the spec helper change intended?
> 
> Could you please clarify what do you mean ?

removing `config.treat_symbols_as_metadata_keys_with_true_values = true` in [spec/spec_helper.rb](https://github.com/shop-ware/s3_direct_upload/pull/1/files/c11a4c92ff1ce6333e54ddaad8de144c1a069989#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94)

**Comment by ghost** [comment]
> > > was the spec helper change intended?
> > 
> > 
> > Could you please clarify what do you mean ?
> 
> removing `config.treat_symbols_as_metadata_keys_with_true_values = true` in [spec/spec_helper.rb](https://github.com/shop-ware/s3_direct_upload/pull/1/files/c11a4c92ff1ce6333e54ddaad8de144c1a069989#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94)

I remove this because it's no need longer, it's enabled for default

File: spec/spec_helper.rb (hunk 1 above)
Comment (colinross) [reply spec/spec_helper.rb:5]: 👍

File: app/assets/javascripts/s3_direct_upload.js.coffee [hunk 2]
```diff
@@ -63,7 +63,7 @@ $.fn.S3Uploader = (options) ->
         data.context.find('.bar').css('width', progress + '%')
 
     done: (e, data) ->
-      content = build_content_object $uploadForm, data.files[0], data.result
+         = build_content_object $uploadForm, data.files[0], data.result
```
Comment (colinross) [inline app/assets/javascripts/s3_direct_upload.js.coffee]: maybe i'm missing something here in coffee syntax, but is this right?

**Comment by colinross** [comment]
@artempopkov0 is this change backward compatible? Do we need to version this?

**Comment by ghost** [comment]
> is this change backward compatible? Do we need to version this?

It is backward compatible. So we don't need to version this, because even with these changes we can still use the gem as before.

File: app/assets/javascripts/s3_direct_upload.js.coffee (hunk 2 above)
Comment (ghost) [reply app/assets/javascripts/s3_direct_upload.js.coffee]: Oh, this is my mistake, i already fixed it

**Review by colinross — APPROVED** [review]
